src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/RenameTypeService.cs

```diff
@@ -0,0 +1,189 @@
+            foreach (EnvDTE.CodeElement element in elements)
+            {
+                // Return the code element if it is a type with the same name
+                if ((element.Kind == EnvDTE.vsCMElement.vsCMElementStruct ||
+                     element.Kind == EnvDTE.vsCMElement.vsCMElementClass ||
+                     element.Kind == EnvDTE.vsCMElement.vsCMElementModule ||
+                     element.Kind == EnvDTE.vsCMElement.vsCMElementInterface ||
+                     element.Kind == EnvDTE.vsCMElement.vsCMElementEnum) &&
+                     SyntaxFactsService.StringComparer.Equals(oldName, element.Name) &&
+                     element is EnvDTE80.CodeElement2 element2)
+                {
+                    codeElementToRename = element2;
+                    return true;
+                }
+
+                // If no matching names were found recurse into namespace-like and type-like definitions
+                else if ((element.Kind == EnvDTE.vsCMElement.vsCMElementNamespace ||
+                          element.Kind == EnvDTE.vsCMElement.vsCMElementClass ||
+                          element.Kind == EnvDTE.vsCMElement.vsCMElementModule ||
+                          element.Kind == EnvDTE.vsCMElement.vsCMElementInterface) &&
+                          TryGetCodeElementToRename(oldName, element.Children, out codeElementToRename))
+                {
+                    return true;
+                }
+            }
+
+            // No elements were found with a matching name
+            codeElementToRename = null;
+            return false;
+        }
+
+        private async Task<ISymbol> TryGetSymbolToRenameAsync(string name,
+                                                              string filePath,
+                                                              CancellationToken cancellationToken = default)
+        {
+            if (!TryGetDocument(filePath, out Document newDocument))
+                return null;
+
+            SyntaxNode root = await newDocument.GetSyntaxRootAsync(cancellationToken);
+            if (root is null)
+                return null;
+
+            SemanticModel semanticModel = await newDocument.GetSemanticModelAsync();
+            if (semanticModel is null)
+                return null;
+
+            IEnumerable<SyntaxNode> declarations = root.DescendantNodes().Where(n => HasMatchingSyntaxNode(semanticModel, n, name, cancellationToken));
+            SyntaxNode declaration = declarations.FirstOrDefault();
+            if (declaration is null)
+                return null;
+
+            return semanticModel.GetDeclaredSymbol(declaration);
+        }
+
+        private bool TryGetDocument(string filePath, out Document document)
+        {
+            // NOTE: It is safe to grab the first project we encounter which
+            // contains the given file. Roslyn will handle the case where the
+            // file is included in multiple projects (linked-files or multi-TFM)
```

Comment:
'In that it will rename the type in the what is considered the "active editor context"'